src/scripts/main.js

```diff
@@ -1,3 +1,386 @@
 'use strict';
 
-// write your code here
+const boardSize = 4;
+const gameBoard = document.querySelector('.game-board');
+const gameScore = document.querySelector('.game-score');
+const button = document.querySelector('.button');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+const handlerSlide = e => {
+  switch (e.code) {
+    case 'ArrowLeft':
+      if (slideLeft()) {
+        setTimeout(() => {
```

Comment:
You will have a problem with this approach. If a user will rapidly presses random keys your timeouts will be stacked. 
It's better to block any interaction while animation continues and a tile creates.

It's some improvement but you could think about it.

Reply:
> You will have a problem with this approach. If a user will rapidly presses random keys your timeouts will be stacked. It's better to block any interaction while animation continues and a tile creates.
> 
> It's some improvement but you could think about it.

Maybe I should implement slide functions as async functions, wait when all tiles will be moved and then create a new tile.